File: 백준/Silver/1912. 연속합/연속합.cc
#include <bits/stdc++.h>

using namespace std;

int n;
int arr[100001];
int dp[100001] = { 0, };
int Max = -9999;

int main()
{
	cin >> n;

	for (int i = 1; i <= n; i++)
	{
		cin >> arr[i];
		dp[i] = arr[i];
	}

	dp[1] = arr[1];

	for (int i = 1; i <= n; i++)
	{
		dp[i] = max(dp[i], dp[i - 1] + arr[i]);

		if (Max < dp[i])
			Max = dp[i];
	}

	cout << Max;

	return 0;
}

File: 백준/Silver/1912. 연속합/연속합.java
import java.util.*;
import java.io.*;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringBuilder sb = new StringBuilder();
    static StringTokenizer st;

    static int n, max;
    static int[] arr, dp;

    public static void main(String[] args) throws IOException {
        n = Integer.parseInt(br.readLine());

        arr = new int[n];
        dp = new int[n];

        st = new StringTokenizer(br.readLine());
        for (int i = 0 ; i < n ; i ++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        dp[0] = arr[0];
        max = arr[0];

        for (int i = 1 ; i < n ; i ++) {
            dp[i] = Math.max(dp[i-1] + arr[i], arr[i]);
            if (dp[i] > max) max = dp[i];
        }

        System.out.println(max);

    }
}
